x-pack/plugins/security_solution/public/common/components/ml/hooks/use_installed_security_jobs.test.ts

```diff
@@ -0,0 +1,99 @@
+import { useAppToasts } from '../../../hooks/use_app_toasts';
+import { useAppToastsMock } from '../../../hooks/use_app_toasts.mock';
+import { mockJobsSummaryResponse } from '../../ml_popover/api.mock';
+import { getJobsSummary } from '../api/get_jobs_summary';
+import { useInstalledSecurityJobs } from './use_installed_security_jobs';
+
+jest.mock('../../../../../common/machine_learning/has_ml_user_permissions');
+jest.mock('../../../../../common/machine_learning/has_ml_license');
+jest.mock('../../../hooks/use_app_toasts');
+jest.mock('../api/get_jobs_summary');
+
+describe('useInstalledSecurityJobs', () => {
+  let appToastsMock: jest.Mocked<ReturnType<typeof useAppToastsMock.create>>;
+
+  beforeEach(() => {
+    appToastsMock = useAppToastsMock.create();
+    (useAppToasts as jest.Mock).mockReturnValue(appToastsMock);
+    (getJobsSummary as jest.Mock).mockResolvedValue(mockJobsSummaryResponse);
+  });
+
+  describe('when the user has permissions', () => {
+    beforeEach(() => {
+      (hasMlUserPermissions as jest.Mock).mockReturnValue(true);
+      (hasMlLicense as jest.Mock).mockReturnValue(true);
+    });
+
+    it('returns jobs and permissions', async () => {
+      const { result, waitForNextUpdate } = renderHook(() => useInstalledSecurityJobs());
+      await waitForNextUpdate();
+
+      expect(result.current.jobs).toHaveLength(3);
+      expect(result.current.jobs).toEqual(
+        expect.arrayContaining([
+          {
+            datafeedId: 'datafeed-siem-api-rare_process_linux_ecs',
+            datafeedIndices: ['auditbeat-*'],
+            datafeedState: 'stopped',
+            description: 'SIEM Auditbeat: Detect unusually rare processes on Linux (beta)',
+            earliestTimestampMs: 1557353420495,
+            groups: ['siem'],
+            hasDatafeed: true,
+            id: 'siem-api-rare_process_linux_ecs',
+            isSingleMetricViewerJob: true,
+            jobState: 'closed',
+            latestTimestampMs: 1557434782207,
+            memory_status: 'hard_limit',
+            processed_record_count: 582251,
+          },
+        ])
+      );
+      expect(result.current.isMlUser).toEqual(true);
+      expect(result.current.isLicensed).toEqual(true);
+    });
+
+    it('filters out non-security jobs', async () => {
+      const { result, waitForNextUpdate } = renderHook(() => useInstalledSecurityJobs());
+      await waitForNextUpdate();
+
+      expect(result.current.jobs.length).toBeGreaterThan(0);
+      expect(result.current.jobs.every(isSecurityJob)).toEqual(true);
+    });
+
+    it('renders a toast error if the ML call fails', async () => {
+      (getJobsSummary as jest.Mock).mockRejectedValue('whoops');
+      const { waitForNextUpdate } = renderHook(() => useInstalledSecurityJobs());
+      await waitForNextUpdate();
+
+      expect(appToastsMock.addError).toHaveBeenCalledWith('whoops', {
+        title: 'Security job fetch failure',
+      });
```

Comment:
We've had a gap in our coverage around the ML hooks for a long time, so awesome to see these tests! 🚀 🎉